src/controller/python/chip/yaml/parser.py

```diff
@@ -29,11 +29,170 @@
 from chip.yaml.errors import ParsingError, UnexpectedParsingError
 from .data_model_lookup import *
 import chip.yaml.format_converter as Converter
+from .response_storage import ResponseStorage
 
 _SUCCESS_STATUS_CODE = "SUCCESS"
+_NODE_ID_DEFAULT = 0x12345
+_ENDPOINT_DETAULT = ''  # TODO why is this an empty string
+_CLUSTER_DEFAULT = ''
+_TIMEOUT_DEFAULT = 90
 logger = logging.getLogger('YamlParser')
 
 
+@dataclass
+class _ReferenceObjects:
+    # Data model lookup to get python attribute, cluster, command object.
+    data_model_lookup: DataModelLookup = None
+    # Where various test action response are stored and loaded from.
+    response_storage: ResponseStorage = None
+    # Top level configuration values for a yaml test.
+    config_values: dict = None
+
+
+class _ConstraintValue:
+    def __init__(self, value, field_type, reference_objects: _ReferenceObjects):
+        self._response_storage = reference_objects.response_storage
+        self._load_expected_response_key = None
```

Comment:
This looks like a boolean name, but it's a field that relates to a value indirection where instead of a literal, we have an variable indirection where we compare to a previously saved object.

Recommend `self._indirect_value_key`.

Suggest also documenting what this is --> that we are a binding a name to the constraint value, and the actual value can only be looked-up dynamically, which is why this is a key name.